File: backend/app/schemas/cover_letter/cover_letter_analysis.py
from typing import List
from pydantic import BaseModel


class SmartSuggestion(BaseModel):
    title: str
    description: str
    type: str
    priority: str


class RecruiterAnalysis(BaseModel):
    strongest_area: str
    improvement_opportunity: str
    recruiter_impression: str


class CoverLetterAnalysisRequest(BaseModel):
    tone: str
    language: str
    job_posting: str
    subject: str
    body: str


class CoverLetterAnalysisResponse(BaseModel):
    smart_suggestions: List[SmartSuggestion]
    recruiter_analysis: RecruiterAnalysis

File: frontend/src/features/cover-letter/types/coverLetterAnalysis.ts
export interface SmartSuggestion {
  title: string;
  description: string;
  type:
    | "warning"
    | "success"
    | "improvement";
  priority:
    | "high"
    | "medium"
    | "low";
}

export interface RecruiterAnalysis {
  strongest_area: string;
  improvement_opportunity: string;
  recruiter_impression: string;
}

export interface CoverLetterAnalysis {
  smart_suggestions: SmartSuggestion[];

  recruiter_analysis: RecruiterAnalysis;
}
